File: typescript/test/stage5/Stage5Test.java
package org.training.speedrun.stage5;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.assertj.core.api.Assertions.assertThat;

class Stage5Test {

    private final Stage5 stage5 = new Stage5();

    @Test
    void long_words_are_eligible_if_they_have_4_vowels() {
        assertThat(stage5.isEligible("azertyuiopqsdfg")).isTrue();
    }

    @Test
    void long_words_are_eligible_if_they_have_at_least_3_special_letters() {
        assertThat(stage5.isEligible("rzmztypopqsdfg")).isTrue();
    }

    @Test
    void short_words_are_eligible_if_they_have_at_least_5_letters_in_upper_case() {
        assertThat(stage5.isEligible("PlKZMZ")).isTrue();
    }

    @Test
    void short_words_are_eligible_if_they_have_3_vowels() {
        assertThat(stage5.isEligible("aeio")).isTrue();
    }

    @ParameterizedTest
    @ValueSource(strings = {"pazof", "AOJGKFLDLFMG", "", "oplm", "lkoirp"})
    void other_words_are_not_elligible(String word) {
        assertThat(stage5.isEligible(word)).isFalse();
    }
}

File: typescript/test/stage5/Stage5.test.ts
import { expect } from 'chai';
import { Stage5 } from '../../src/stage5/Stage5';

describe('Stage5', () => {
  it('should define long words eligible if they have 4 vowels', () => {
    const stage5 = new Stage5();
    expect(stage5.isEligible('azertyuiopqsdfg')).to.be.true;
  });

  it('should define long words eligible if they have at least 3 special letters', () => {
    const stage5 = new Stage5();
    expect(stage5.isEligible('rzmztypopqsdfg')).to.be.true;
  });

  it('should define short words eligible if they have at least 5 letters in upper case', () => {
    const stage5 = new Stage5();
    expect(stage5.isEligible('PlKZMZ')).to.be.true;
  });

  it('should define short words eligible if they have 3 vowels 4 vowels', () => {
    const stage5 = new Stage5();
    expect(stage5.isEligible('aeio')).to.be.true;
  });

  const assertions: string[] = ['pazof', 'AOJGKFLDLFMG', '', 'oplm', 'lkoirp'];
  assertions.forEach(function(word) {
    it('should not define eligible the following other words: ' + word, () => {
      const stage5 = new Stage5();
      expect(stage5.isEligible(word)).to.be.false;
    });
  });
});
